compiler/fir/providers/src/org/jetbrains/kotlin/fir/types/InferenceUtils.kt

```diff
@@ -112,16 +112,16 @@ fun ConeKotlinType.suspendFunctionTypeToFunctionTypeWithContinuation(session: Fi
     val kind =
         if (isKFunctionType(session)) FunctionClassKind.KFunction
         else FunctionClassKind.Function
-    val functionalTypeId = ClassId(kind.packageFqName, kind.numberedClassName(typeArguments.size))
     val fullyExpandedType = type.fullyExpandedType(session)
     val typeArguments = fullyExpandedType.typeArguments
+    val functionalTypeId = ClassId(kind.packageFqName, kind.numberedClassName(typeArguments.size))
     val lastTypeArgument = typeArguments.last()
     return ConeClassLikeTypeImpl(
         ConeClassLikeLookupTagImpl(functionalTypeId),
         typeArguments = (typeArguments.dropLast(1) + ConeClassLikeLookupTagImpl(continuationClassId).constructClassType(
             arrayOf(lastTypeArgument),
             isNullable = false
-        ) + lastTypeArgument).toTypedArray(),
+        ) + session.builtinTypes.nullableAnyType.type).toTypedArray(),
```

Comment:
It's not clear for me why this line is changed. Could you explain?

Reply:
Technically the correct return type of suspend functional objects is `Any?` because their `invoke()` method returns an unrepresentable type `T | CoroutineSingletons`. This is also what FE1.0 writes to metadata. I don't think it matters much anyway since deserializer will drop this argument and read the original return type from `Continuation`'s argument.